docs/book/features/vsphere_csi_migration.md

```diff
@@ -0,0 +1,211 @@
+   - `CSIMigrationvSphere` requires `CSIMigration` feature flag to be enabled. This flag is enabling CSI migration on the Kubernetes Cluster.
+
+    Steps:
+   - Enable feature flags `CSIMigration` and `CSIMigrationvSphere` on `kube-controller` and `kubelet` on all primary nodes.
+     - update kube-controller-manager manifest file and following arguments. This file is generally available at `/etc/kubernetes/manifests/kube-controller-manager.yaml`
+
+            `- --feature-gates=CSIMigration=true,CSIMigrationvSphere=true`
+     - update kubelet configuration file and add following flags. This file is generally available at `/var/lib/kubelet/config.yaml`
+
+              featureGates:
+              {
+                "CSIMigration": true,
+                "CSIMigrationvSphere": true
+              }
+
+   - Enable feature flags `CSIMigration` and `CSIMigrationvSphere` on `kubelet` on all workload nodes. Please note that before changing the configuration on the Kubelet on each node we **must drain** the node (remove running application workloads).  
+     - Node drain example.
+
+            $ kubectl drain k8s-node1 --force --ignore-daemonsets
+            node/k8s-node1 cordoned
+            WARNING: deleting Pods not managed by ReplicationController, ReplicaSet, Job, DaemonSet or StatefulSet: default/vcppod; ignoring DaemonSet-managed Pods: kube-system/kube-flannel-ds-amd64-gs7fr, kube-system/kube-proxy-rbjx4, kube-system/vsphere-csi-node-fh9f6
+            evicting pod default/vcppod
+            pod/vcppod evicted
+            node/k8s-node1 evicted
+
+     - After migration is enabled, make sure `csinodes` instance for the node is updated with `storage.alpha.kubernetes.io/migrated-plugins` annotation.
+
+           $ kubectl describe csinodes k8s-node1
+           Name:               k8s-node1
+           Labels:             <none>
+           Annotations:        storage.alpha.kubernetes.io/migrated-plugins: kubernetes.io/vsphere-volume
+           CreationTimestamp:  Wed, 29 Apr 2020 17:51:35 -0700
+           Spec:
+             Drivers:
+               csi.vsphere.vmware.com:
+                 Node ID:  k8s-node1
+           Events:         <none>
+
+   - Repeat this for all workload nodes in the Kubernetes Cluster.
+
+6. There is also an optional `CSIMigrationvSphereComplete` flag that can be enabled if all the nodes have CSI migration enabled. `CSIMigrationvSphereComplete` helps stop registering the vSphere in-tree plugin in kubelet and volume controllers and enables shims and translation logic to route volume operations from the vSphere in-tree plugin to vSphere CSI plugin. Requires `CSIMigration` and `CSIMigrationvSphere` feature flags enabled and vSphere CSI plugin installed and configured on all nodes in the cluster.
```

Comment:
Requires `CSIMigration` and `CSIMigrationvSphere` feature flags enabled and vSphere CSI plugin installed and configured on all nodes in the cluster. - can we elaborate on this?

Reply:
clarified

> `CSIMigrationvSphereComplete` flag requires `CSIMigration` and `CSIMigrationvSphere` feature flags enabled and vSphere CSI plugin installed and configured on all nodes in the cluster.